src/neuroconv/datainterfaces/ecephys/generatorrecordinginterface.py

```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from .baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ...utils.dict import dict_deep_update
+
+
+class GeneratorRecordingInterface(BaseRecordingExtractorInterface):
+    """An interface for testing purposes."""
+
+    def __init__(
+        self,
+        num_channels=4,
+        sampling_frequency=30_000.0,
+        durations=[1.0],
+        seed=0,
+        verbose=True,
+        es_key: str = "ElectricalSeries",
+    ):
+        from spikeinterface.core.generate import generate_recording
+
```

Comment:
I am rather against having this be an actual DataInterface stored in the main listing - can you move it to the `tools.testing.mock_interfaces` and name it `MockRecordingInterface` instead?

Reply:
What do you mean in the main listing?

This would not show in datainterfaces because is not in /src/neuroconv/datainterfaces/__init__.py`, correct?

I put it here because it depends on spikeinterface and I remember that we wanted to segragate interfaces by dependencies. That said, having it in `tools.testing.mock_interfaces` makes sense to me as well if you think that's better.

As for the naming, `MockRecordingInterface` works for me. One important aspect is that after the next version of spikeinterface is out this will be a proper generator. That is, it will have a recorder with as much data as desired but loaded lazily. This can be used to tests chunked writing and memory tests in general. That's why I put the name Generator in the name.

See:
https://github.com/SpikeInterface/spikeinterface/issues/1413

Reply:
By 'listing' I mean anyone perusing the package structure (either via deep imports or on GitHub) of the ecephys interfaces, which are supposed to be format/file type specific

That all sounds great, the parameters of the generation should definetely be exposed to the `__init__` so we can make a `MockRecordingInterface` take on any behavior we wish to test to mimic a real `RecordingInterface`

If you want to name it `MockGenereratorRecordingInterface` or `GeneratorRecordingInterface`, they should be fine - I'd just favor `Mock` starting the title due to precedence of the other recently added cases for things like this

Reply:
Makes sense. Let's go with the simplest option and later when the new version I will point out the property in the docstring.